src/commands/Minion/alch.ts

```diff
@@ -43,21 +44,33 @@ export default class extends BotCommand {
 	@minionNotBusy
 	@requiresMinion
 	async run(msg: KlasaMessage, [quantity = null, item]: [number | null, Item[]]) {
-		const userBank = msg.author.settings.get(UserSettings.Bank);
-		const osItem = item.find(i => userBank[i.id] && i.highalch && i.tradeable);
-		if (!osItem) {
-			return msg.channel.send("You don't have any of this item to alch.");
+		const userBank = msg.author.bank();
+		let osItem = item?.find(i => userBank.has(i.id) && i.highalch && i.tradeable);
+
+		const [favAlchs] = msg.author.settings
+			.get(UserSettings.FavoriteAlchables)
+			.filter(id => userBank.has(id))
+			.map(getOSItem)
+			.filter(i => i.highalch > 0)
+			.sort((a, b) => b.highalch - a.highalch);
```

Comment:
this is duplicated in the agility alching isnt it? might be better to put this in a utility function if so, e.g. `getUsersFavAlchs` or something.